### DISCUSSION TITLE
Add Activity To Component

### DISCUSSION THREAD

**Discussion by QuangLanh** [clientIO/joint#2454] [Q&A]
### Introduction

How can I add + and - buttons like this to the component. When I click on the + button, it will add that component to my table, and when I click -, it will delete that component from the table mine ?

### Steps to reproduce

![add or delete to table](https://github.com/clientIO/joint/assets/91791283/22ec6d71-0387-4a5f-a660-33281a03488b)

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
I assume the buttons need to be visible only when the element is selected,
You can add [custom handles](https://resources.jointjs.com/docs/rappid/v3.7/ui.Halo.html#ui.Halo.customizing-tools) to the `ui.Halo`, or you can add the [elementTools.Button](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#elementTools.Button).

**Comment by QuangLanh** [comment]
Thanks I finished it
